diff --git a/configure b/configure
index e242e73..73ac65c 100755
--- a/configure
+++ b/configure
@@ -621,6 +621,8 @@ if [[ $host_os = mingw* || $host_os = msys* || $host_os = cygwin* ]]; then
         elif cpp_check '' '' 'defined(_M_ARM)' ; then
             host_cpu=arm
         fi
+    elif [[ "$cc_base" = clang || "$cc_base" = clang[\ .]* ]]; then
+        compiler=CLANG
     else
         # MinGW uses broken pre-VS2015 Microsoft printf functions unless it's told to use the POSIX ones.
         CFLAGS="$CFLAGS -D_POSIX_C_SOURCE=200112L"
@@ -1633,7 +1635,7 @@ includedir=$includedir
 Name: x264
 Description: H.264 (MPEG4 AVC) encoder library
 Version: $(grep POINTVER < x264_config.h | sed -e 's/.* "//; s/".*//; s/ .*//')
-Libs: -L$libdir -lx264 $([ "$shared" = "yes" ] || echo $libpthread $libm $libdl)
+Libs: -L$libdir -llibx264 $([ "$shared" = "yes" ] || echo $libpthread $libm $libdl)
 Libs.private: $([ "$shared" = "yes" ] && echo $libpthread $libm $libdl)
 Cflags: -I$includedir $([ "$shared" = "yes" ] && echo "-DX264_API_IMPORTS")
 EOF
